### DISCUSSION TITLE
How to set Metadata while storing a file using Storage?

### DISCUSSION THREAD

**Discussion by TheArKaID** [laravel/framework#53588] [Q&A]
Asking here because [thephpleague/flysystem](https://github.com/thephpleague/flysystem) and [thephpleague/flysystem-aws-s3-v3](https://github.com/thephpleague/flysystem-aws-s3-v3) is not opening Discussion.

Is it possible to include metadata when storing a file? As far as I can find, [AWS S3](https://docs.aws.amazon.com/AmazonS3/latest/userguide/UsingMetadata.html) and [Minio](https://min.io/docs/minio/linux/developers/javascript/API.html#putObject) support this metadata insertion. How to do it in Laravel? Can't find any docs about it.

**Comment by denjaland** [comment]
I don't think it's supported out of the box, but you could probably do it with the AWS SDK after uploading, or just extend the base adapter and extend it with the support you need?
